oras/auth/token.py

```diff
@@ -100,6 +107,8 @@ def request_token(self, h: auth_utils.authHeader) -> bool:
         """
         params = {}
         headers = {}
+        if self._basic_auth:  # we exchange the basic auth for the token
```

Comment:
Wouldn't this be a separate part of basic auth?

Reply:
My understanding of this section of the token backend is:
- We want to exchange a basic auth for an auth token
- to do so, we need to supply our basic auth to the realm
- the realm, seeing our basic auth credential, will give us an auth bearer token (which we could use in the subsequent requests)

so haven't fully understood how you would rewrite this part? 🤔 can you expand it a little bit more for me please?

Reply:
Gotcha - I think I am wondering what the failure case here is. If we didn't get an anon or authenticated token, then the auth backend remains basic. Should we allow that or fail somewhere with a targeted message?

Reply:
mmm... the reason for this changed line is different: this is method for requiring auth-token, but was not providing any auth, it was requesting a token which --for all intent and purposes-- was analogous to the anon-token requested by the other method.

I could verify in my case the experienced behaviour was:
- from provider, 1st request (no auth at all) failed
- from provider, delegate to auth backend to perform authentication
- the token auth backend, requested anon token
- the token auth backend, receives from server "I won't give you any anon token"
- the token auth backend, moves on to request an auth-token
- _before this code change_ the token auth backed did not provide any basic auth to the Realm, so in effect it was analogous to the anon token just failed --because servers such as ghcr.io do not offer anon tokens
- _after this code change_ the token auth backend provides the basic auth to the Realm, and the Realm provides back the auth-token successfully

This is my understanding on how the flow is meant to happen, hence why this line change.
If I missed something, kindly let me know!